cli/cmd/deploy/deploy.go

```diff
@@ -276,6 +236,26 @@ func projectParamPrompt(ctx context.Context, c *client.Client, orgName, githubUR
 	return opts, nil
 }
 
+func credentialsPrompt(projectPath string) map[string]string {
+	csMap, err := inferSourceConnectors(projectPath)
+	if err != nil {
+		exitWithFailure(err)
+	}
+
+	creds := make(map[string]string)
+	for connector, sources := range csMap {
```

Comment:
To go closer to the proposal, I think we should decouple the required connectors from the sources. Imagine a project with 10 sources all pulling from S3. So best not to mention the source names.